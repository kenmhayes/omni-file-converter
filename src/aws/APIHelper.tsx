import { API } from 'aws-amplify';

const SESSION_API_NAME = 'SessionGatewayAPI';

export interface CreateSessionRequest {
  conversionRequests: FileConversionInput[];
}

export interface FileConversionInput {
  fileName: string;
  originalFileFormat: string;
  convertedFileFormat: string;
}

interface CreateSessionResponse {
  sessionId: string;
  fileNameToId: { [key: string]: string };
}

export interface GetSessionResponse {
  expirationTime: number;
  conversions: ConversionOutput[];
}

export interface ConversionOutput {
  id: string;
  fileName: string;
  conversionStatus: string;
  originalFileFormat: string;
  convertedFileFormat: string;
}

// eslint-disable-next-line max-len
export const createSession = async (request: CreateSessionRequest): Promise<CreateSessionResponse> => API.post(
  SESSION_API_NAME,
  '/createsession',
  {
    body: request,
  },
);

export const getSession = async (sessionId: string): Promise<GetSessionResponse> => API.get(
  SESSION_API_NAME,
  '/getsession',
  {
    queryStringParameters: {
      sessionid: sessionId,
    },
  },
);
