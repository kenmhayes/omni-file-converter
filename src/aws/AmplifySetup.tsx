import { Amplify, Auth, Storage } from 'aws-amplify';

function AmplifySetup() {
  /**
 * Initialize Cognito authentication
 */
  Amplify.configure({
    Auth: {
      identityPoolId: process.env.REACT_APP_IDENTITY_POOL_ID,
      region: process.env.REACT_APP_AWS_REGION,
    },
    Storage: {
      AWSS3: {
        bucket: process.env.REACT_APP_S3_BUCKET,
        region: process.env.REACT_APP_AWS_REGION,
      },
    },
  });

  Auth.configure();

  const getCredentials = async () => {
    await Auth.currentCredentials();
  };

  getCredentials();

  // Configure S3 storage
  Storage.configure({
    customPrefix: {
      public: '',
    },
  });
}

export default AmplifySetup;
