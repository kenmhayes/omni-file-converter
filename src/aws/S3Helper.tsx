import { Storage } from 'aws-amplify';

export const getS3ObjectUrl = async (objectKey: string): Promise<string> => Storage.get(
  `${process.env.REACT_APP_S3_CONVERTED_PREFIX}${objectKey}`,
);

export const putS3Object = async (
  objectKey: string,
  convertType: string,
  file: File,
): Promise<void> => {
  await Storage.put(`${process.env.REACT_APP_S3_ORIGINAL_PREFIX}${convertType}/${objectKey}`, file);
};
