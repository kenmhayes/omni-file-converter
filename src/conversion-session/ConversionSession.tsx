import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import { Alert, Spinner } from 'react-bootstrap';
import { ConversionOutput, GetSessionResponse, getSession } from '../aws/APIHelper';
import ConversionList from './ConversionList';
import icons from '../assets/svg/bootstrap-icons.svg';

interface LoadedBodyProps {
  expirationTime: number;
  conversions: ConversionOutput[];
}

function PreloadBody() {
  return <Spinner animation="border" />;
}

function LoadedBody(props: LoadedBodyProps) {
  const { expirationTime, conversions } = props;

  const readableDate = new Date(expirationTime * 1000).toDateString();

  return (
    <div>
      <Alert variant="primary" className="d-flex align-items-center">
        <svg className="me-2" width="16" height="16" fill="currentColor">
          <use xlinkHref={`${icons}#info-circle`} />
        </svg>
        <div>
          Your converted files are available until
          {' '}
          <strong><em>{readableDate}</em></strong>
          . Save this link if you wish to download them at a later time.
        </div>

      </Alert>
      <ConversionList conversions={conversions} />
    </div>
  );
}

function ConversionSession() {
  const { sessionId } = useParams();
  const [session, setSession] = useState<GetSessionResponse>();

  const fetchData = async (id: string) => {
    const response = await getSession(id);
    setSession(response);
  };

  useEffect(() => {
    fetchData(sessionId as string);
  }, [sessionId]);

  return (
    <div>
      {
        session
          ? <LoadedBody expirationTime={session.expirationTime} conversions={session.conversions} />
          : <PreloadBody />
      }
    </div>
  );
}

export default ConversionSession;
