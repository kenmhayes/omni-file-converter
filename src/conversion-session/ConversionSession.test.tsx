import React from 'react';
import { act, create } from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';
import * as APIHelper from '../aws/APIHelper';
import ConversionSession from './ConversionSession';
import { MemoryRouter, Route, Routes } from 'react-router-dom';

const apiSpy = jest.spyOn(APIHelper, 'getSession');

describe('ConversionSession', () => {
    it('renders a conversion session', async () => {
        let component= create(
            <MemoryRouter initialEntries={["/1234"]}>
                <Routes>
                    <Route path="/:sessionId" element={<ConversionSession/>}>
                    </Route>
                </Routes>
            </MemoryRouter>
        );;

        expect(component.toJSON()).toMatchSnapshot();
    });

    it('loads session data after API call', async () => {
        apiSpy.mockResolvedValue({
            expirationTime: 188888888,
            conversions: [
                {
                    fileName: 'test1.jpg',
                    id: '123',
                    conversionStatus: 'COMPLETED',
                    originalFileFormat: 'jpg',
                    convertedFileFormat: 'png',
                }
            ]
        });
        let component = (<MemoryRouter initialEntries={["/1234"]}>
                            <Routes>
                                <Route path="/:sessionId" element={<ConversionSession/>}>
                                </Route>
                            </Routes>
                    </MemoryRouter>);
        const { findByRole, rerender } = render(component);

        expect(await findByRole("alert")).toBeInTheDocument();
        expect(apiSpy).toHaveBeenCalledWith("1234");
    });
});

