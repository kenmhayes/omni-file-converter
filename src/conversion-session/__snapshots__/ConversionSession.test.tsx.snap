// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversionSession renders a conversion session 1`] = `
<div>
  <div
    className="spinner-border"
  />
</div>
`;
